source/agora/node/Ledger.d

```diff
@@ -1118,15 +1130,18 @@ public class Ledger
 
         Returns:
             true if the validator has revealed its preimage for the provided
-                block height
+                block height or if current node
 
     ***************************************************************************/
 
     private bool hasRevealedPreimage (in Height height, in Hash utxo_key)
         @safe nothrow
     {
         if (utxo_key == this.enroll_man.getEnrollmentKey())
+        {
+            log.info("As this utxo key is ours we will assume preimage is revealed");
```

Comment:
Idempotent functions like this should not do logging

Reply:
Updated